### PR TITLE
Add bulk edit sample UI tests

### PR DESCRIPTION
**Pull request by callunity**

### CODE REVIEW

**Comment by sqwci** [comment]
SonarQube analysis reported 4 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [Dtos.java#L594](https://github.com/TGAC/miso-lims/blob/29351ba6d8e5c148e9967d24c8c748d28b2fa60f/miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/Dtos.java#L594): Refactor this method to reduce its Cognitive Complexity from 17 to the 15 allowed. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)
1. ![CRITICAL][CRITICAL] [Dtos.java#L1704](https://github.com/TGAC/miso-lims/blob/29351ba6d8e5c148e9967d24c8c748d28b2fa60f/miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/Dtos.java#L1704): Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1186)
1. ![MAJOR][MAJOR] [Dtos.java#L1704](https://github.com/TGAC/miso-lims/blob/29351ba6d8e5c148e9967d24c8c748d28b2fa60f/miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/Dtos.java#L1704): Remove this unused method parameter "from". [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR][MAJOR] [hot_sample.js#L357](https://github.com/TGAC/miso-lims/blob/29351ba6d8e5c148e9967d24c8c748d28b2fa60f/miso-web/src/main/webapp/scripts/hot_sample.js#L357): Rename "i" as this name is already used in declaration at line 349. [rule](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS2814)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/Dtos.java [hunk 1]
```diff
@@ -1726,6 +1702,7 @@ public static Run to(NotificationDto from, User user) {
   }
 
   private static void setPacBioRunValues(PacBioNotificationDto from, PacBioRun to) {
```
Comment (sqwci) [inline miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/Dtos.java]: MAJOR Remove this unused method parameter "from". [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1172)

**Comment by sqwci** [comment]
SonarQube analysis reported 2 issues
* ![CRITICAL][CRITICAL] 1 critical
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [Dtos.java#L594](https://github.com/TGAC/miso-lims/blob/249535136eb0b4a78599e46a7282f7d5721ad93a/miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/Dtos.java#L594): Refactor this method to reduce its Cognitive Complexity from 17 to the 15 allowed. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by sqwci** [comment]
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [Dtos.java#L592](https://github.com/TGAC/miso-lims/blob/9d1c5e6470970dfb424d25579f0b0fae084350a6/miso-dto/src/main/java/uk/ac/bbsrc/tgac/miso/dto/Dtos.java#L592): Refactor this method to reduce its Cognitive Complexity from 17 to the 15 allowed. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)
1. ![MAJOR][MAJOR] [hot_sample.js#L357](https://github.com/TGAC/miso-lims/blob/9d1c5e6470970dfb424d25579f0b0fae084350a6/miso-web/src/main/webapp/scripts/hot_sample.js#L357): Rename "i" as this name is already used in declaration at line 349. [rule](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS2814)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: miso-web/src/main/webapp/scripts/hot_sample.js [hunk 2]
```diff
@@ -175,7 +175,7 @@ HotTarget.sample = (function() {
               numberOfMonths : 1
             },
             allowEmpty : true,
-            include : true,
+            include : !Constants.isDetailedSample || (Constants.isDetailedSample && config.targetSampleClass.alias != 'Identity'),
```
Comment (apmasell) [inline miso-web/src/main/webapp/scripts/hot_sample.js]: The second `Constants.isDetailedSample` check is redundant.

**Comment by sqwci** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [hot_sample.js#L357](https://github.com/TGAC/miso-lims/blob/4408f2b591a51d1b0a36d0cc1d5de2a8857a34d0/miso-web/src/main/webapp/scripts/hot_sample.js#L357): Rename "i" as this name is already used in declaration at line 349. [rule](https://sonarcloud.io/coding_rules#rule_key=javascript%3AS2814)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: miso-web/src/it/java/uk/ac/bbsrc/tgac/miso/webapp/integrationtest/SampleBulkITUtils.java [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+import uk.ac.bbsrc.tgac.miso.core.data.impl.SampleStockImpl;
+import uk.ac.bbsrc.tgac.miso.core.data.impl.SampleTissueImpl;
+import uk.ac.bbsrc.tgac.miso.core.data.impl.SampleTissueProcessingImpl;
+import uk.ac.bbsrc.tgac.miso.core.util.LimsUtils;
+import uk.ac.bbsrc.tgac.miso.webapp.integrationtest.page.BulkSamplePage.Columns;
+import uk.ac.bbsrc.tgac.miso.webapp.integrationtest.page.element.HandsOnTable;
+import uk.ac.bbsrc.tgac.miso.webapp.integrationtest.page.element.HandsOnTableSaveResult;
+import uk.ac.bbsrc.tgac.miso.webapp.integrationtest.page.element.SampleHandsOnTable;
+
+public class SampleBulkITUtils extends AbstractIT {
```
Comment (djcooke) [inline miso-web/src/it/java/uk/ac/bbsrc/tgac/miso/webapp/integrationtest/SampleBulkITUtils.java]: Make abstract and maybe rename to `AbstractBulkSampleIT` or something similar. `XUtils` classes usually contain public static utility methods

**Pull request merged**